docs/api/ethereum.md

```diff
@@ -1,5 +1,645 @@
+
+### assetLedgerSource
+
+A class instance `variable` holding a `string` which represents the URL to the compiled ERC-721 related smart contract definition file. This file is used when deploying new asset ledgers to the network.
+
+### emit(event, ...options);
+
+A `synchronous` class instance `function` to manually trigger a provider event.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| options | [required] Pass valid current and previous account ID for `ACCOUNT_CHANGE` event or valid current and previous network version for `NETWORK_CHANGE` event.
+
+**Result:**
+
+An instance of the same provider class.
+
+**Example:**
+
+```ts
+import { AccountEvent } from '@0xcert/ethereum-bitski-backend-provider';
+
+mutation.emit(AccountEvent.NETWORK_CHANGE, '3');
+```
+
+### getAvailableAccounts()
+
+An `asynchronous` class instance `function` which returns currently available Ethereum wallet addresses.
+
+**Result:**
+
+A list of `strings` representing Ethereum account IDs.
+
+**Example:**
+
+```ts
+// perform query
+const accountIds = await provider.getAvailableAccounts();
+```
+
+### getInstance(options)
+
+A static class `function` that returns a new instance of the BitskiProvider class (alias for `new BitskiProvider`).
+
+**Arguments**
+
+See the class [constructor](#bitski-provider) for details.
+
+**Usage**
+
+```ts
+import { BitskiProvider } from '@0xcert/ethereum-bitski-backend-provider';
+
+// create provider instance
+const provider = BitskiProvider.getInstance();
+```
+
+### getNetworkVersion()
+
+An `asynchronous` class instance `function` which returns Ethereum network version (e.g. `1` for Ethereum Mainnet).
+
+**Result:**
+
+A `string` representing Ethereum network version.
+
+**Example:**
+
+```ts
+// perform query
+const version = await provider.getNetworkVersion();
+```
+
+### isCurrentAccount(accountId)
+
+A `synchronous` class instance `function` which returns `true` when the provided `accountId` maches the currently set account ID.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| accountId | [required] A `string` representing Ethereum account address.
+
+**Result:**
+
+A `boolean` which tells if the `accountId` maches the currently set account ID.
+
+**Example:**
+
+```ts
+// ethereum wallet address
+const walletId = '0x06012c8cf97bead5deae237070f9587f8e7a266d';
+
+// perform query
+const maches = provider.isCurrentAccount(walletId);
+```
+
+### isSupported()
+
+A `synchronous` class instance `function` which returns `true` when the provider is supported by the environment.
+
+**Result:**
+
+A `boolean` which tells if the provider can be used.
+
+**Example:**
+
+```ts
+// perform query
+const isSupported = provider.isSupported();
+```
+
+### isUnsafeRecipientId(ledgerId)
+
+A `synchronous` class instance `function` which returns `true` when the provided `id` is listed among unsafe recipient ids on the provided.
```

Comment:
`A `synchronous` class instance `function` which returns `true` when the provided `id` is listed among unsafe recipient IDs on the provider.`